provisioning/miniprov/hare_mp/main.py

```diff
@@ -445,10 +445,36 @@ def start_mkfs_parallel(hostname: str, hare_config_dir: str):
                       f'{perf_result}\n\n')
 
 
+@repeat_if_fails()
+def wait_for_mkfs_to_complete_on_all_nodes(utils: Utils,
+                                           cns_utils: ConsulUtil):
+    nodes = utils.get_io_nodes()
+    for node in nodes:
+        while not cns_utils.kv.kv_get(f'mkfs_done/{node}', recurse=True):
```

Comment:
As far as I understand, if mkfs fails to finish at some remote node, another node which is blocked on `wait_for_mkfs_to_complete_on_all_nodes` will hang forever. Is it acceptable?

We could have introduced some inter-nodes communication protocol using KV as a channel so that failures would also be reflected there and thus the local mkfs failure will affect other init containers. But I wonder if it is worth doing.

@mssawant, @madhavemuri Do you have any thoughts?

Reply:
I think this being only at bootstrap, we can either add additional code and fail or let remote processes wait. In both the cases the problem will be reported. Let's hold on to this change for now, and take it as an improvement if required. what do you think?

Reply:
Sounds fine to me. It can be just another PR :-)

Reply:
Agree, some way of identifying the failures and breaking the deployment will help.

Reply:
Failure will be surely reported for a given node, just that other nodes would wait as one of the node did not complete. But failure will be reported for sure.